### PR TITLE
Add type definitions to webpack-plugin-serve

### PR DESCRIPTION
**Pull request by matheus1lva**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present). 

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/webpack-plugin-serve/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    };
+    compress?: boolean;
+    historyFallback?: boolean | HistoryApiFallbackOptions;
+    hmr?: boolean;
+    host?: string | Promise<string>;
+    http2?: boolean | Http2ServerOptions | Http2SecureServerOptions;
+    https?: HttpsServerOptions;
+    liveReload?: boolean;
+    log?: {
+        level: string;
```
Comment (sheetalkamat) [inline types/webpack-plugin-serve/index.d.ts]: level should be of type `'trace' | 'debug' | 'info' | 'warn' | 'error'`

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/webpack-plugin-serve/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    match: RegExpMatchArray;
+    parsedUrl: Url;
+}
+
+export interface Rewrite {
+    from: RegExp;
+    to: string | RegExp | RewriteTo;
+}
+
+export interface HistoryApiFallbackOptions {
```
[the remark this answers is not in the document]
Comment (PlayMa256) [reply types/webpack-plugin-serve/index.d.ts]: As far as i can tell, those options aren't exported from the types package. As per https://unpkg.com/@types/connect-history-api-fallback@1.3.1/index.d.ts

[the remark this answers is not in the document]
Comment (sheetalkamat) [reply types/webpack-plugin-serve/index.d.ts]: You can make them `public` since its just interface

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Still should reuse the type from HistoryFallback instead of duplicating

File: types/webpack-plugin-serve/index.d.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (PlayMa256) [reply types/webpack-plugin-serve/index.d.ts]: I actually am planning to edit the connect-history-api-fallback to export everything and use on this PR.

How should i proceed? Open a new pr updating the other package, than update this one? Or putting everything in the same PR?

[the remark this answers is not in the document]
Comment (sheetalkamat) [reply types/webpack-plugin-serve/index.d.ts]: You can have it in same PR

**Comment by PlayMa256** [comment]
@sheetalkamat All done! I hope this is good now!
Let me know if you need me to change anything else!

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**